src/gpu-compute/compute_unit.cc

```diff
@@ -409,8 +409,6 @@ ComputeUnit::doInvalidate(RequestPtr req, int kernId){
 
     // kern_id will be used in inv responses
     gpuDynInst->kern_id = kernId;
-    // update contextId field
-    req->setContext(gpuDynInst->wfDynId);
```

Comment:
Does this get set somewhere else now?

Reply:
No. As far as I can tell this is not needed.

Reply:
I'm not sure why it goes with this commit then?  But I guess ok ...

Reply:
Things found with asan/ubsan

Reply:
My point was more should it be a separate commit/pull request.  But small enough that probably does not matter.

Reply:
It is it's own commit, there are 3 in this PR

Reply:
gotcha ... this is something I find much harder to conceptualize in Github vs. Gerrit